agent/exec/dockerapi/executor.go

```diff
@@ -11,12 +11,14 @@ import (
 	"github.com/docker/swarmkit/api"
 	"github.com/docker/swarmkit/log"
 	"golang.org/x/net/context"
+	"sync"
 )
 
 type executor struct {
 	client           engineapi.APIClient
 	secrets          exec.SecretsManager
 	genericResources []*api.GenericResource
+	mutex            *sync.Mutex // This mutex protects the following node field
```

Comment:
The field doesn't have to be a pointer. This can be something like:

```go
mu             sync.Mutex
```

and then it doesn't need to be initialized.